lang/llvm-6.0/Portfile

```diff
@@ -584,6 +584,29 @@ if {${subport} eq "llvm-${llvm_version}"} {
         }
     }
 
+    variant emulated_tls description { enable c11/c++11 thread support on older systems using emulated-tls } {
+        # allow clang to use emulated-tls to support thread_local on systems prior to 10.7
+        # emulated TLS works when linking against macports-libstdc++ or against
+        # libc++ / libc++abi with cxa_thread_atexit and emutls.c objects included
+        patchfiles-append \
+            9000-patch-clang-6.0-support-emulated-tls.diff
+    }
+    if {${os.major} < 11} {
+        default_variants-append +emulated_tls
+    }
+
+
+    variant defaultlibcxx description {default to -stdlib=libc++ on all systems}  {
+        #patch clang to always default to -stdlib=libc++ if not otherwise specified
+        patchfiles-append 9003-patch-clang-6.0-Toolchains-Darwin-default-to-libcxx-on-all-systems.diff
+    }
+
+    # if macports.conf is configured to stdlib=libc++ then make that the default if not otherwise specified
+    # exactly matches the behaviour of newer systems
+    if { ${cxx_stdlib} eq "libc++" } {
```

Comment:
I think I should add a test here for the ${os.version}, so that the +defaultlibcxx variant isn't added to all the newer systems that don't actually need this. It doesn't change any functionality, but keeps the variant list cleaner for all the new systems. I'll do that later on today.